ghost/admin/app/helpers/activity-feed-fetcher.js

```diff
@@ -0,0 +1,141 @@
+import moment from 'moment-timezone';
+import {Resource} from 'ember-could-get-used-to-this';
+import {TrackedArray} from 'tracked-built-ins';
+import {action} from '@ember/object';
+import {inject as service} from '@ember/service';
+import {task} from 'ember-concurrency';
+import {tracked} from '@glimmer/tracking';
+
+const actions = {
+    showPrevious: 'showPrevious',
+    showNext: 'showNext'
+};
+
+export default class MembersEventsFetcher extends Resource {
+    @service ajax;
+    @service ghostPaths;
+    @service store;
+
+    @tracked data = new TrackedArray([]);
+    @tracked isLoading = false;
+    @tracked isError = false;
+    @tracked errorMessage = null;
+
+    @tracked hasReachedStart = true;
+    @tracked hasReachedEnd = true;
+
+    @tracked shownEvents = 0;
+    @tracked totalEvents = 0;
+
+    // save last event's date of each page for easy navigation to previous page
+    @tracked eventsBookmarks = [];
+
+    get value() {
+        return {
+            isLoading: this.isLoading,
+            isError: this.isError,
+            errorMessage: this.errorMessage,
+            data: this.data,
+            loadNextPage: this.loadNextPage,
+            loadPreviousPage: this.loadPreviousPage,
+            hasReachedStart: this.hasReachedStart,
+            hasReachedEnd: this.hasReachedEnd,
+            totalEvents: this.totalEvents,
+            shownEvents: this.shownEvents,
+            previousEvents: this.getAmountOfPreviousEvents()
+        };
+    }
+
+    getAmountOfPreviousEvents() {
+        return this.totalEvents > this.args.named.pageSize
+            ? this.shownEvents - this.args.named.pageSize + 1
+            : this.data.length;
+    }
+
+    async setup() {
+        const currentTime = moment.utc().format('YYYY-MM-DD HH:mm:ss');
+        let filter = `data.created_at:<'${currentTime}'`;
+
+        if (this.args.named.filter) {
+            filter += `+${this.args.named.filter}`;
+        }
+
+        await this.loadEventsTask.perform({filter}, actions.showNext);
+    }
+
+    @action
+    loadNextPage() {
+        const lastEvent = this.data[this.data.length - 1];
+        const lastEventDate = moment.utc(lastEvent.data.created_at).format('YYYY-MM-DD HH:mm:ss');
+        let filter = `data.created_at:<'${lastEventDate}'`;
+
+        if (this.args.named.filter) {
+            filter += `+${this.args.named.filter}`;
+        }
+
+        this.eventsBookmarks.push(lastEventDate);
+        this.loadEventsTask.perform({filter}, actions.showNext);
+    }
+
+    @action
+    loadPreviousPage() {
+        this.eventsBookmarks.pop();
+        let filter = `data.created_at:<'${this.eventsBookmarks[this.eventsBookmarks.length - 1]}'`;
```

Comment:
Do we need to push the initial `currentTime` to the bookmarks in `setup`? I would think that going to the second page, then back to the previous page would let `this.eventsBookmarks[this.eventsBookmarks.length - 1]` evaluate to undefined 🤔

Reply:
We need `currentTime` just to keep first element of every page. For example, when we go to the next page we'll have 2 elements in `eventsBookmarks`.
How they appeared:
- User open first page, we save time of first event of first page, `this.eventsBookmarks = ['firstPageEvent']`
- User open second page, we save time of first event of second page, `this.eventsBookmarks = ['firstPageEvent', 'secondPageEvent']`

When user go to the previous one, we need first element of previous page. So, in this case, we go to the first page (`this.eventsBookmarks[0]`)

Quite messy, but didn't find better solution 😅

Reply:
You say that `['firstPageEvent']` should be in the `this.eventsBookmarks` initially (which is indeed what I meant), but I don't see the code that does that. 
```
    async setup() {
        const currentTime = moment.utc().format('YYYY-MM-DD HH:mm:ss');
        let filter = `data.created_at:<'${currentTime}'`;

        if (this.args.named.filter) {
            filter += `+${this.args.named.filter}`;
        }

        await this.loadEventsTask.perform({filter}, actions.showNext);
    }
  ```
 This is called after the Resource is loaded, right? But nothing is pushed to `eventsBookmarks`